pkg/cache/cache.go

```diff
@@ -7,6 +7,14 @@ import (
 	"github.com/spf13/viper"
 )
 
+type CacheType string
+
+const (
+	Azure     CacheType = "azure"
+	S3        CacheType = "s3"
+	FileBased CacheType = "file"
+)
```

Comment:
Shouldn't we handle cache providers in the same way we handle analyzers/filters?

```go
var remoteCacheProviderMap = map[string]ICache{
	"Azure":  &S3Cache{},
	"Azure": &AzureCache{},
        "File": &FileBasedCache{},
}

func ListRemoteCacheProviders() []string {
	remoteCacheProviderKeys := make([]string, 0, len(remoteCacheProviderMap))
	for k := range remoteCacheProviderMap {
		remoteCacheProviderKeys = append(remoteCacheProviderKeys, k)
	}
	return remoteCacheProviderKeys
}
```

Reply:
a) I think there is not a big value out of having this as an interface, b) the list functionality in caches is listing files from your already configured cache backend and not a list of supported caches from.